sql/core/src/main/scala/org/apache/spark/sql/execution/LocalTableScanExec.scala

```diff
@@ -37,7 +37,7 @@ case class LocalTableScanExec(
     "numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"))
 
   @transient private lazy val unsafeRows: Array[InternalRow] = {
-    if (rows.isEmpty) {
+    if (rows == null || rows.isEmpty) {
```

Comment:
Adding transient is to avoid serializing and copying whole rows, so I guess it's preferred one.

Reply:
Related commits are here:
https://github.com/apache/spark/commit/256358f66a975aa62eb81378e4e073fc2d15644b
https://github.com/apache/spark/commit/f70f46d1e5bc503e9071707d837df618b7696d32

Reply:
Ah, I checked related code and the fix looks reasonable to me.

Reply:
Ah, I see. I'm a bit worried that this value could be different between a driver and executors. Which code touches this variable in executors? What's the stacktrace for `unsafeRows`?

Reply:
Uh, that's actually a kind of defensive programming, as it would bring NPE if there's no check for null. I haven't seen any actual stack trace to refer to `unsafeRows`.

Reply:
I personally think `unsafeRows` is not expected to be called in executor sides, so we need to check `assert(rows == null)` instead of the current approache.